zcommit.py

```diff
@@ -27,6 +33,95 @@
 formatter = logging.Formatter(fmt='%(levelname)-8s %(asctime)s %(message)s')
 handler.setFormatter(formatter)
 
+def format_rename(old, new):
+    prefix = posixpath.commonprefix([old, new])
+    suffix = posixpath.commonprefix([old[::-1], new[::-1]])[::-1]
+    return "%s{%s => %s}%s" % (
+        prefix,
+        old[len(prefix):-len(suffix)],
+        new[len(prefix):-len(suffix)],
+        suffix,
+    )
+
+def format_commit(c, commit_url):
+    info = {'name' : c['author']['name'],
+            'email' : c['author']['email'],
+            'message' : c['message'],
+            'timestamp' : dateutil.parser.parse(c['timestamp']).strftime('%F %T %z'),
+            'url' : c['url']}
+
+    header = """%(url)s
+Author: %(name)s <%(email)s>
+Date:   %(timestamp)s
+
+%(message)s
+---
+""" % info
+
+    try:
+        # Check if the diff is small enough
+        r = requests.get(commit_url, headers={'Accept': 'application/vnd.github.diff'})
```

Comment:
This fetches an arbitrary URL that somebody posted to zcommit, right? This seems not great if it can be arbitrarily large. Should we make sure it's at least a GitHub URL or something?

Reply:
Not sure the best way to do that, so I just checked that it starts with "https://api.github.com/". That'll break if they ever move their API, I suppose.